Client/Frontend/Home/ActivityStreamTopSitesCell.swift

```diff
@@ -0,0 +1,393 @@
+                self.contentView.backgroundColor = UIColor.lightGrayColor()
+                return
+            }
+
+            // Get dominant colors using a scaled 25/25 image.
+            img.getColors(CGSize(width: 25, height: 25)) { colors in
+                //In cases where the background is white. Force the background color to a different color
+                var bgColor: UIColor
+                if colors.backgroundColor.isWhite {
+                    let colorArr = [colors.detailColor, colors.primaryColor].filter { !$0.isWhite }
+                    if colorArr.isEmpty {
+                        bgColor = UIColor.lightGrayColor()
+                    } else {
+                        bgColor = colorArr.first!
+                    }
+                } else {
+                    bgColor = colors.backgroundColor
+                }
+                self.contentView.backgroundColor = bgColor
+            }
+        }
+    }
+
+    func configureWithTopSiteItem(site: TopSiteItem) {
+        titleLabel.text = site.urlTitle
+        if let favURL = site.faviconURL {
+            setImageWithURL(favURL)
+        } else {
+            imageView.image = FaviconFetcher.getDefaultFavicon(site.siteURL)
+        }
+    }
+
+}
+
+struct ASHorizontalScrollCellUX {
+    static let TopSiteCellIdentifier = "TopSiteCell"
+    static let TopSiteItemSize = CGSize(width: 100, height: 100)
+    static let BackgroundColor = UIColor.whiteColor()
+    static let PageControlRadius: CGFloat = 3
+    static let PageControlSize = CGSize(width: 30, height: 15)
+    static let PageControlOffset = -5
+}
+
+/*
+ The View that describes the topSite cell that appears in the tableView.
+ */
+class ASHorizontalScrollCell: UITableViewCell {
+
+    lazy private var collectionView: UICollectionView = {
+        let layout  = HorizontalFlowLayout()
+        layout.itemSize = ASHorizontalScrollCellUX.TopSiteItemSize
+        let collectionView = UICollectionView(frame: CGRectZero, collectionViewLayout: layout)
+        collectionView.registerClass(TopSiteCell.self, forCellWithReuseIdentifier: ASHorizontalScrollCellUX.TopSiteCellIdentifier)
+        collectionView.backgroundColor = ASHorizontalScrollCellUX.BackgroundColor
+        collectionView.showsHorizontalScrollIndicator = false
+        collectionView.pagingEnabled = true
+        return collectionView
+    }()
+
+    lazy private var pageControl: FilledPageControl = {
+        let pageControl = FilledPageControl()
+        pageControl.tintColor = UIColor.grayColor()
+        pageControl.indicatorRadius = ASHorizontalScrollCellUX.PageControlRadius
+        pageControl.userInteractionEnabled = false
+        return pageControl
+    }()
+
+    weak var delegate: ASHorizontalScrollSource? {
+        didSet {
+            collectionView.delegate = delegate
+            collectionView.dataSource = delegate
+            delegate!.pageChangedHandler = { [weak self] progress in
```

Comment:
If we set the delegate to `nil` this will burn the house down 😟 Probably safer to use `?`